packages/toolkit/src/api.ts

```diff
@@ -0,0 +1,15 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Auth, Connection } from 'aws-core-vscode/auth'
+/**
+ * Exposing listConnections API for other extension to read or re-use
+ * the available connections in aws toolkit.
+ */
+export const awsToolkitApi = {
+    async listConnections(): Promise<Connection[]> {
+        return Auth.instance.listConnections()
```

Comment:
Returning the internal `Connection` object is not how this should be exposed. There should be an explicit interface, to avoid accidentally exposing internal things. It may also need to be serializable (not sure)?

> If there is a need for other extensions to use the toolkit connection, the API for getCredentials can be exposed later.

`Connection` already exposes `getCredentials`: https://github.com/aws/aws-toolkit-vscode/blob/7c77f0ec164c906e4a298f09604a77e7e6f26434/packages/core/src/auth/connection.ts#L109-L112

We should not be returning internal objects directly. The public API should only expose things we explicitly want to expose. And in particular, the reason for separating `listConnections` from `getCredentials` is so that we _don't_ have to build an authorization model around `listConnections`, while leaving the door open for `getCredentials` in the future (which may require an authorization model, e.g. asking the user to "confirm that extension X is allowed to use credentials Y").